### PR TITLE
Disables logging in CI to catch issues

### PR DESCRIPTION
**Pull request by PureW**
with ununsed variables

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -66,6 +66,7 @@ jobs:
             && cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_COVERAGE=On -DCPACK_GENERATOR=DEB \
                 -DENABLE_COMPILER_WARNINGS=On -DENABLE_WERROR=Off -DCMAKE_EXPORT_COMPILE_COMMANDS=On \
                 -DCMAKE_CXX_FLAGS="-fuse-ld=lld" \
+                -DLOGGING_LEVEL=NONE \
```
[the remark this answers is not in the document]
Comment (SvetlanaBayarovich) [reply .circleci/config.yml:69]: Yeah, it's just an investigation for now. Trying to understand how many cases are there

[the remark this answers is not in the document]
Comment (PureW) [reply .circleci/config.yml:69]: `TRACE` added to the release build below

**Comment by PureW** [comment]
Hmm, I did not expect to get ASAN errors with this change

```

**Comment by PureW** [comment]
Hmm, I did not expect to get ASAN errors with this change

```
==16348==ERROR: AddressSanitizer: global-buffer-overflow on address 0x7f437a734048 at pc 0x7f438c540a6d bp 0x7ffcd02bab40 sp 0x7ffcd02ba2e8
READ of size 1 at 0x7f437a734048 thread T0
    #0 0x7f438c540a6c  (/lib/x86_64-linux-gnu/libasan.so.5+0x67a6c)
    #1 0x7f43782459f5 in valhalla::odin::Maneuver::ToString[abi:cxx11]() const (/root/project/build/src/libvalhalla.so.3+0x1099f9f5)
    #2 0x7f4377efed55 in valhalla::odin::ManeuversBuilder::Build[abi:cxx11]() (/root/project/build/src/libvalhalla.so.3+0x10658d55)
    #3 0x7f4377d673ad in valhalla::odin::DirectionsBuilder::Build(valhalla::Api&) (/root/project/build/src/libvalhalla.so.3+0x104c13ad)
    #4 0x7f43783c3531 in valhalla::odin::odin_worker_t::narrate(valhalla::Api&) const (/root/project/build/src/libvalhalla.so.3+0x10b1d531)
    #5 0x7f43794e8f44 in valhalla::tyr::actor_t::route(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<void ()> const*, valhalla::Api*) (/root/project/build/src/libvalhalla.so.3+0x11c42f44)
    #6 0x7f438b2f957c in valhalla::gurka::do_action(valhalla::Options_Action const&, valhalla::gurka::map const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<valhalla::baldr::GraphReader>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*) (/root/project/build/test/libvalhalla_test.so+0xc9357c)
    #7 0x7f438b2fb833 in valhalla::gurka::do_action(valhalla::Options_Action const&, valhalla::gurka::map const&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > const&, std::shared_ptr<valhalla::baldr::GraphReader>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (/root/project/build/test/libvalhalla_test.so+0xc95833)
    #8 0x55ef58aafc8f in GuidanceViews_Signboards_CheckGuidanceViews_Test::TestBody() (/root/project/build/test/gurka/gurka_guidance_views_signposts+0x2bec8f)
    #9 0x7f43673ea8db in void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (/root/project/build/lib/libgtest.so+0x5c58db)
    #10 0x7f4367337999 in testing::Test::Run() (/root/project/build/lib/libgtest.so+0x512999)
    #11 0x7f43673381af in testing::TestInfo::Run() (/root/project/build/lib/libgtest.so+0x5131af)
    #12 0x7f4367338a45 in testing::TestSuite::Run() (/root/project/build/lib/libgtest.so+0x513a45)
    #13 0x7f436733b698 in testing::internal::UnitTestImpl::RunAllTests() (/root/project/build/lib/libgtest.so+0x516698)
    #14 0x7f436733d8b4 in testing::UnitTest::Run() (/root/project/build/lib/libgtest.so+0x5188b4)
    #15 0x7f43678862ff in main (/root/project/build/lib/libgtest_main.so+0x12ff)
    #16 0x7f4365f7d0b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #17 0x55ef58aabc7d in _start (/root/project/build/test/gurka/gurka_guidance_views_signposts+0x2bac7d)

0x7f437a734048 is located 56 bytes to the left of global variable '*.LC226' defined in '/root/project/src/odin/maneuver.cc' (0x7f437a734080) of size 10
  '*.LC226' is ascii string ' base_id='
0x7f437a734048 is located 1 bytes to the right of global variable '*.LC225' defined in '/root/project/src/odin/maneuver.cc' (0x7f437a734040) of size 7
  '*.LC225' is ascii string ' type='
SUMMARY: AddressSanitizer: global-buffer-overflow (/lib/x86_64-linux-gnu/libasan.so.5+0x67a6c) 
Shadow bytes around the buggy address:
  0x0fe8ef4de7b0: 00 00 00 00 f9 f9 f9 f9 00 00 00 00 06 f9 f9 f9
  0x0fe8ef4de7c0: f9 f9 f9 f9 00 00 00 00 01 f9 f9 f9 f9 f9 f9 f9
  0x0fe8ef4de7d0: 00 00 f9 f9 f9 f9 f9 f9 00 01 f9 f9 f9 f9 f9 f9
  0x0fe8ef4de7e0: 00 f9 f9 f9 f9 f9 f9 f9 00 00 07 f9 f9 f9 f9 f9
  0x0fe8ef4de7f0: 00 00 03 f9 f9 f9 f9 f9 02 f9 f9 f9 f9 f9 f9 f9
=>0x0fe8ef4de800: 00 02 f9 f9 f9 f9 f9 f9 07[f9]f9 f9 f9 f9 f9 f9
  0x0fe8ef4de810: 00 02 f9 f9 f9 f9 f9 f9 00 06 f9 f9 f9 f9 f9 f9
  0x0fe8ef4de820: 02 f9 f9 f9 f9 f9 f9 f9 02 f9 f9 f9 f9 f9 f9 f9
  0x0fe8ef4de830: 00 00 00 05 f9 f9 f9 f9 00 00 00 00 00 00 05 f9
  0x0fe8ef4de840: f9 f9 f9 f9 00 00 00 00 02 f9 f9 f9 f9 f9 f9 f9
  0x0fe8ef4de850: 00 00 00 00 00 00 02 f9 f9 f9 f9 f9 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==16348==ABORTING
```

**Comment by PureW** [comment]
Ahh, looks like `src/odin/maneuver.cc` conditionally adds code when `TRACE` logging is in effect.  Likely means that some of that code has issues which have been hidden before when not enabled.

**Comment by mandeepsandhu** [comment]
> Ahh, looks like `src/odin/maneuver.cc` conditionally adds code when `TRACE` logging is in effect. Likely means that some of that code has issues which have been hidden before when not enabled.

Yep, its printing out properties of a maneuver object. Some places where its logging the attributes, its trying to reserve space in the string which might not be sufficient for holding all the data we're putting within it. I think the reserve can be removed as this code is only enabled for debugging and its okay to be slow and not pre-reserve the string space.

**Comment by kevinkreiser** [comment]
@PureW that is exactly the kind of thing i was hoping we would find but adding TRACE :smile: normally its a flat out compiler error, ASAN is good to catch too though!

**Comment by dgearhart** [comment]
@PureW the update was missed when the type was switched to an enum in this [PR](https://github.com/valhalla/valhalla/pull/2687)

**Comment by PureW** [comment]
@SvetlanaBayarovich @kevinkreiser All green now!

File: src/odin/maneuver.cc [hunk 2]
```diff
@@ -1279,7 +1279,7 @@ std::string Maneuver::ToString() const {
   for (const auto& guidance_view : guidance_views_) {
     man_str += "[";
     man_str += " data_id=" + guidance_view.data_id();
-    man_str += " type=" + guidance_view.type();
+    // man_str += " type=" + guidance_view.type();
```
Comment (kevinkreiser) [inline src/odin/maneuver.cc:1282]: @dgearhart do you want to write a function that converts this to a string or is it ok for us to just omit it?

Comment (mandeepsandhu) [reply src/odin/maneuver.cc:1282]: we can also just use `std:to_string` on it? (it might be helpful to have the guidance view type logged)

Comment (dgearhart) [reply src/odin/maneuver.cc:1282]: agreed - we can just do `std:to_string` for now and add function later

Comment (PureW) [reply src/odin/maneuver.cc]: Pushed change, all green again

**Review by kevinkreiser — APPROVED** [review]

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**